contrib/olivere/elastic/elastictrace.go

```diff
@@ -98,7 +100,7 @@ func quantize(url, method string) string {
 // from to access the entire data including the snippet. max is used to specify the length
 // of the stream contained in the reader. If unknown, it should be -1. If 0 < max < n it
 // will override n.
-func peek(rc io.ReadCloser, max int, n int) (string, io.ReadCloser, error) {
+func peek(rc io.ReadCloser, gzipped bool, max, n int) (string, io.ReadCloser, error) {
```

Comment:
Instead of having a boolean argument, let's have:
```go
func peek(rc io.ReadCloser, encoding string, max, n int) (string, io.ReadCloser, error) {
```
And check if `encoding == "gzip"`